### DISCUSSION TITLE
client.connected "Connect internal"

### DISCUSSION THREAD

**Discussion by MAKOMO** [pymodbus-dev/pymodbus#2325] [Q&A] [answered]
The property `connected` of the (sync) serial client tries to connect if not connected. I understand that this is the interpretation of its documentation "Connect internal", while on the (sync) tcp client it only returns the connection status (so it does not "Connect internal"). Is there a specific reason for this semantic difference? Asking from the perspective of an app supporting both, MODBUS serial and tcp communication.

**Comment by janiversen** [comment]
The sync code is very old code, and we only touch it when really needed, because it is (slowly) being replaced by a common implementation.

Sounds like you found a bug, feel free to submit a PR.

**Reply by MAKOMO** [reply]
What is the intended behavior and automatic reconnect or just the return of the connection status. The documentation "Connect internal" is unclear to me.

**Reply by janiversen** [reply]
Try to connect if not connected, just like with any request.

**Reply by MAKOMO** [reply]
Now I am even more confused. `connected` is a property not a request, right? Shouldn't it just return the connection status without changing it?

**Reply by janiversen** [reply]
why ? every api call that involves communication tries to connect.

But anyhow, the code is as it is, but as usual PRs are welcome.

**Reply by MAKOMO** [reply]
Funny, this issue was discussed last year already (Discussion #1854). I now submitted a PR #2327 correcting the implementation of that connected property and its documentation. Note that to check if the socket exists and is the method `open is_socket_open` should be used. 


Note that the `connected` property [is not used within the lib](https://github.com/search?q=repo%3Apymodbus-dev%2Fpymodbus%20connected&type=code).

**Comment by janiversen** [comment, marked as answer]
PR merged, thanks.
